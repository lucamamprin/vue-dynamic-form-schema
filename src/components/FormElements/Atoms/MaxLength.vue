<template functional>
  <div class="text-right text-light">
    <span
      :id="props.srCharacterCount"
      class="sr-only"
    >
      {{ `${props.typedtext ? 0 : props.typedText.length} characters on ${props.maxLength} possible for ${props.label}` }}
    </span>

    <p class="m-0">
      {{ props.typedText.length }} / {{ props.maxLength }}
    </p>
  </div>
</template>
