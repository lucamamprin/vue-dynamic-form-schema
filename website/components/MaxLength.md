# MaxLength

## Props

<!-- @vuese:MaxLength:props:start -->
|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|srCharacterCount|-|`String`|`true`|-|
|typedText|-|`String`|`false`|-|
|maxLength|-|`Number`|`true`|-|
|label|-|`String`|`true`|-|

<!-- @vuese:MaxLength:props:end -->


