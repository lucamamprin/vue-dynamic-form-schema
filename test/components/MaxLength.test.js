import { shallowMount } from "@vue/test-utils"
import MaxLength from "../../src/components/FormElements/Atoms/MaxLength.vue"

const propsDefault = {
  typedText: "",
  maxLength: 50,
  label: "First name",
  srCharacterCount: "first-name-counter",
}

let wrapper

describe("max length indicator", () => {

  beforeEach(() => {
    wrapper = shallowMount(MaxLength, {
      propsData: propsDefault,
    })
  })

  test("mounts correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
