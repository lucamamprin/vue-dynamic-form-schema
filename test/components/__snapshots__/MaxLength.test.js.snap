// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`max length indicator mounts correctly 1`] = `
<div class="text-right text-light"><span id="first-name-counter" class="sr-only">
    0 characters on 50 available for First name
  </span>
  <p class="m-0">
    0 / 50
  </p>
</div>
`;
